barretenberg/cpp/src/barretenberg/translator_vm/translator_circuit_builder.cpp

```diff
@@ -356,47 +338,25 @@ TranslatorCircuitBuilder::AccumulationInput TranslatorCircuitBuilder::generate_w
 
     return input;
 }
-/**
- * @brief Create a single accumulation gate
- *
- * @param acc_step
- */
-void TranslatorCircuitBuilder::create_accumulation_gate(const AccumulationInput acc_step)
+
+void TranslatorCircuitBuilder::assert_well_formed_accumulation_input(const AccumulationInput& acc_step)
```

Comment:
My goal here was to separate the sanity checking from main logic of the code as it is quite hard to discern the actual logic of `create_accumulation_gate` from boilerplate code. Another small change was to move the anonymous function to appear right before they are being used to reduce the back and forth for the reader - ofc way more work needs to be done here.